lib/linguist/heuristics.yml

```
@@ -841,6 +841,12 @@ disambiguations:
   - language: JSON
     pattern: '\"modelName\"\:\s*\"GM'
   - language: Yacc
+- extensions: ['.tsp']
+  rules:
+  - language: TypeSpec
+    pattern: '^(import|using|namespace|interface|op|model|scalar|alias|union|enum)\s'
+  - language: TravelingSalesmanProblem
```

```suggestion
  - language: TSPLIB data
```

Following the change requested (and explained) below.